packages/blade/src/components/Dropdown/Dropdown.tsx

```diff
@@ -40,8 +37,14 @@ function Dropdown({ children, selectionType = 'single' }: DropdownProps): JSX.El
   const [shouldIgnoreBlur, setShouldIgnoreBlur] = React.useState(false);
   const selectInputRef = React.useRef<HTMLButtonElement>(null);
   const actionListRef = React.useRef<HTMLDivElement>(null);
-  const dropdownBaseId = useId('dropdown');
+  const [hasFooterAction, setHasFooterAction] = React.useState(false);
+  const [optionsRecalculateToggle, setOptionsRecalculateToggle] = React.useState(false);
```

Comment:
didn't understand what is this used for?

Reply:
removed it. will explain over call